### DISCUSSION TITLE
多样式文本如何实现

### DISCUSSION THREAD

**Discussion by 0x1306a94** [Tencent/libpag#1840] [General] [answered]
如下图所示，想要第一个人民币符号字号比后面文本的字号小，同时又需要整体能够自动居中。
应该怎么实现，是在 AE中有啥方式可以实现。还是需要使用代码设置matrix。
<img width="357" alt="截屏2023-11-07 20 02 46" src="https://github.com/Tencent/libpag/assets/14822396/ab39140d-61f4-4a32-99c5-a2d3a0cf9e1c">

**Comment by kevingpqi123** [comment, marked as answer]
两种实现方式：
（1）设计成两个文本图层，分别进行编辑，然后通过 setMatrix 调整位置信息
（2）富文本通过平台的文本组件来实现，然后截图，通过占位图填充的方式添加到 pag 中

**Reply by 0x1306a94** [reply]
@kevingpqi123 我是在服务端环境使用，所以只能通过第一种方式了吧
第一种方式具体的做法：先替换两个文本图层的内容，然后获取两个文本图层的getBounds 再计算出合适的matrix 对吧

**Comment by 0x1306a94** [comment]
@kevingpqi123 我通过代码动态添加一个文本图层，然后 getBounds 是zero。
```C++
auto symbolLayer = pag::PAGTextLayer::Make(m_file->duration(), "¥");
m_file->addLayer(symbolLayer);
auto symbolRect = symbolLayer->getBounds();
```

**Reply by kevingpqi123** [reply]
这里仅仅设置了一个文本进去，没有执行渲染，肯定是空的哈，需要执行 PAGPlayer 的 flush 方法，然后再获取
PAGLayer 上面的 getBounds 接口获取的仅仅是相对于父级的，如果需要获取到根节点的 bounds 信息，需要通过 PAGPlayer 的 getBounds 接口

**Comment by 0x1306a94** [comment]
@kevingpqi123 嗯呢。不调用flush的话，需要通过  PAGPlayer 的 getBounds 接口 也能拿到。
另外再咨询个问题，就是我有一个一个文本图层，然后AE里面设置的是在画布的中心对齐。
如果我想在代码中通过matrix 重新调整位置。应该是怎样的流程哇？没有找到C++的代码示例
